Create cluster object and plotting

catterbu opened the pull request:
The cluster object uses kmeans clustering and silhouette scores to determine optimal clustering. The plotting functions take this object and create plots such as a WSS plot, gap analysis plot, silhouette plot, and cluster membership box plot.
- [x] rename cluster object
- [x] add `tryCatch`

catterbu committed:
Initial cluster object commit
  - Add initial parameters
  - Add Roxygen skeleton

catterbu committed:
Start cluster object
  - Add import cluster
  - Add check that krange does not include 1
  - Create distance object from data
  - Write lapply for km

catterbu committed:
Add cluster:: to silhouette function call

catterbu committed:
Add for loop to build most of model object
  - Build km_obj
  - Build sil_avg object
  - Build num_clusters object
  - Build sil silhouette object

catterbu committed:
Add structure to make S3

catterbu committed:
Start cluster_obj documentation
  - Add title
  - Add description
  - Add parameters

catterbu committed:
Add class argument to structure call

catterbu committed:
Finish initial clust_obj documentation
  - Complete description
  - Add return section
  - Add seealso section
  - Add examples section

catterbu committed:
Specify 1 cannot be in krange in cluster_obj documentation

catterbu committed:
Add .Rd documentation files compiled

catterbu committed:
Initial cluster plotting commit
  - Add wss_plot initial parameters
  - Add wss_plot roxygen skeleton

catterbu committed:
Modify names of objects returned in cluster_obj

catterbu committed:
Add code to generate plot
  - Create wss using vapply
  - Plot cluster wss
  - Add conditional for optimal clustering

catterbu committed:
Add clusGap object and document it

catterbu committed:
Remove extra spaces documentation

catterbu committed:
Add clusGap_pot
  - Add roxygen skeleton
  - Add gap analysis plot wrapper
  - Add conditional to plot optimal clusters

catterbu committed:
Add pca_plot parameters and documentation skeleton

catterbu committed:
Add pca_plot
  - Create princomp object
  - Plot first two components colored by cluster

catterbu committed:
Add sil_plot parameters and documentation skeleton

catterbu committed:
Add sil_plot
  - Extract silhouette object from clust_obj
  - Plot extracted silhouette object

catterbu committed:
Add avg_sil_plot parameters and documentation skeleton

catterbu committed:
Add avg_sil_plot
  - Extract average silhouette from clust_obj and plot
  - Add optimal option

catterbu committed:
Clarify documentation of parameters and adjust defaults for them

catterbu committed:
Access named list and object elements with brackets instead of $

catterbu committed:
Change name from cluster_obj to cluster_optimal

catterbu committed:
Change determination optimal clustering for given k to withinss

catterbu committed:
Modify wss_plot
  - Extract wss from clust_obj
  - Find start and end of krange
  - Subset out NAs

catterbu committed:
Modify avg_sil_plot
  - Find start and end of krange
  - Subset out NAs
  - Adjust if optimal accordingly

catterbu committed:
Change name in class creation to match new function name

catterbu committed:
Add ellipsis argument to plot functions to be passed to plot

catterbu committed:
Minor code fixes and typos
  - Fix references to outside functions in documentation
  - Fix typos in cluster_obj documentation
  - Fix cluster_optimal typo in example in documentation

catterbu committed:
Create documentation for cluster plotting function

catterbu committed:
Add explicit package reference to clust_boxplot

catterbu committed:
Build .Rd file documentation for cluster functions

catterbu committed:
Add method parameter to cluster_optimal for later use

catterbu committed:
Change single bracket to double bracket in many functions
  - Change kmm wss extraction from single to double brackets
  - Change km_opt cluster extraction from single to double quotes
  - Change clust_gap, k_best, and which.max to double brackets
  - Change the double brackets for length in clusGap function
  - Fix wss_plot brackets and optimal plotting
  - Fix clusGap_plot brackets and optimal plotting
  - Fix pca_plot brackets
  - Fix sil_plot brackets and modify plot title
  - Fix avg_sil_plot brackets and optimal plotting
  - Fix clust_boxplot brackets

catterbu committed:
Minor fixes to pass lintr in travis build
  - Explicitly reference ggplot2 with facet and scale_fill
  - Removed no longer used variables sil_max and sil_avg_max
  - Change cluster_optimal single quotes to double
  - Change wss_plot single quotes to double
  - Change clusGap_plot single quotes to double
  - Change pca_plot single quotes to double
  - Change sil_plot single quotes to double
  - Change avg_sil_plot single quotes to double
  - Change clust_boxplot single quotes to double

catterbu committed:
Change cluster_optimal to multiClust in all references

catterbu committed:
Reference kmeans with stats::kmeans

catterbu committed:
Change data parameter to d in cluster_obj and plotting

nsh87 commented:
There were some errors when checking the built package on this PR, but I wasn't detecting them properly. It's fixed now, which is why your build check now fails. The fixes went into dev, so just rebase off of dev again before pushing any fixes. I think that should do it for the changes into dev for a bit (until we merge this PR).

catterbu committed:
Rebuild documentation after minor roxygen tweaks

catterbu committed:
Add imports back in and update NAMESPACE

catterbu committed:
Rename cluster_obj to multi_clustering

catterbu committed:
Change function name to multi_clust and fix all references

catterbu committed:
Update NAMESPACE after fixing DESCRIPTION

catterbu committed:
Update documentation based on roxygen comments

catterbu committed:
Add validation to clustering functions
  - Add validate_num_data internal function
  - Use validate_not_null and num_data in multi_clust
  - Add validate_sort_range and use in multi_clust
  - Add validate_pos_num and use in multi_clust
  - Add check to validate_num_data for only 0 and 1
  - Move validate_pos_num from multi_cluster to arg_validation
  - Add validate_multi_clust to multi_clustering
  - Add validation to cluster_plotting

catterbu committed:
Fix build errors
  - Add missing validate_sort_range curly brace
  - Fix incorrect length call in validate_sort_range
  - Fix class check in validate_pos_num
  - Add checks for integer and positive in validate_pos_num

catterbu committed:
Add documentation for new validation functions

catterbu committed:
Add details section and modify seealso for clustering functions
  - Add details section and modify seealso for multi_clust
  - Add details section and modify seealso for wss_plot
  - Add details section and modify seealso for clusGap_plot
  - Add details section and modify seealso for avg_sil_plot
  - Add details section and modify seealso for sil_plot
  - Add details section and modify seealso for pca_plot
  - Add details section and modify seealso for clust_boxplot

catterbu committed:
Add documentation changes made in roxygen to .Rd files

catterbu commented:
@nsh87 If it passes, everything should be ready.

nsh87 commented:
Awesome plots! Finally got a chance to run them all. Really nice job, and the documentation is so much more helpful and comprehensive than 99% of the documentation I've ever seen. I opened a couple issues, we can fix them before v1. When you're fixing them just close them in your commit messages with `Fixes #11` to close Issue #11, for example. The issues will be closed automatically when the commit gets merged into master.
